src/api/ingest/serializers.py

```diff
@@ -0,0 +1,511 @@
+        "Management *": "data__sample_event__management",
+        "Sample date: Year *": "data__sample_event__sample_date__year",
+        "Sample date: Month *": "data__sample_event__sample_date__month",
+        "Sample date: Day *": "data__sample_event__sample_date__day",
+        "Sample time": "data__sample_event__sample_time",
+        "Depth *": "data__sample_event__depth",
+        "Interval size": "data__interval_size",
+        "Visibility": "data__sample_event__visibility",
+        "Current": "data__sample_event__current",
+        "Relative depth": "data__sample_event__relative_depth",
+        "Tide": "data__sample_event__tide",
+        "Notes": "data__sample_event__notes",
+        "Observer emails *": "data__observers",
+        "Observation interval *": "data__obs_benthic_pits__interval",
+    }
+
+    # CHOICES
+    _choices = ChoiceViewSet().get_choices()
+    visibility_choices = build_choices("visibilities", _choices)
+    current_choices = build_choices("currents", _choices)
+    relative_depth_choices = build_choices("relativedepths", _choices)
+    tide_choices = build_choices("tides", _choices)
+
+    _reverse_choices = {}
+
+    # PROJECT RELATED CHOICES
+    project_choices = {
+        "data__sample_event__site": None,
+        "data__sample_event__management": None,
+    }
+
+    # FIELDS
+    id = serializers.UUIDField(format="hex_verbose")
+    stage = serializers.IntegerField()
+    project = serializers.UUIDField(format="hex_verbose")
+    profile = serializers.UUIDField(format="hex_verbose")
+    data__protocol = serializers.CharField(required=True, allow_blank=False)
+
+    data__sample_event__site = serializers.CharField()
+    data__sample_event__management = serializers.CharField()
+    data__sample_event__sample_date = serializers.DateField()
+    data__sample_event__sample_time = serializers.TimeField(default="00:00:00")
+    data__sample_event__depth = serializers.DecimalField(max_digits=3, decimal_places=1)
+
+    data__sample_event__visibility = serializers.ChoiceField(
+        choices=visibility_choices, required=False, allow_null=True, allow_blank=True
+    )
+    data__sample_event__current = serializers.ChoiceField(
+        choices=current_choices, required=False, allow_null=True, allow_blank=True
+    )
+    data__sample_event__relative_depth = serializers.ChoiceField(
+        choices=relative_depth_choices,
+        required=False,
+        allow_null=True,
+        allow_blank=True,
+    )
+    data__sample_event__tide = serializers.ChoiceField(
+        choices=tide_choices, required=False, allow_null=True, allow_blank=True
+    )
+    data__sample_event__notes = serializers.CharField(required=False, allow_blank=True)
+
+    # data__observers = ProjectProfileSerializer(many=True)
+    data__observers = serializers.ListField(
+        child=serializers.CharField(), allow_empty=False
+    )
+
+    class Meta:
+        list_serializer_class = CollectRecordCSVListSerializer
```

Comment:
interesting... so how is `many_init` forced to be always true? If I called a subclass of this without `many=True` what would happen?

Reply:
Added to README that it must be True